ocamltest/result.ml

```diff
@@ -55,3 +55,28 @@ let is_pass r = r.status = Pass
 let is_skip r = r.status = Skip
 
 let is_fail r = r.status = Fail
+
+module Join = struct
+  (* The sequential join passes if both tests pass.
+
+     This implies that a linear sequence of actions, a path along the
+     test tree, is considered succesful if all actions passed. *)
+  let sequential r1 r2 =
+    match r1.status, r2.status with
+    | Fail, _ | _, Fail -> r1
```

Comment:
Surely this should be `Fail, _ -> r1 | _, Fail -> r2`? We don't want `Pass, Fail -> Pass` ?